src/components/LoginForm.js

```diff
@@ -0,0 +1,142 @@
+import { useState, useEffect, useReducer, useContext } from "react"
+import { Link, useMatch, useResolvedPath, useNavigate } from "react-router-dom"
+import  {UserContext}  from "./UserContext.js"
+
+import axios from "axios"
+
+
+
+
+function LoginForm(){
+
+    // Boolean state to check if the user is logged in or not
+    const [logged, setLogged] = useState(false)
+    // State to store users
+    const [user, setUser] = useState({name:"", password: ""})
+    // State to store already registered users
+    const [registeredusers, setRegisteredUsers] = useState([]) 
+    // State to store if the user is trying to log in or register
+    const [formaction, setFormAction] = useState("")
+    // navigation
+    const navigate = useNavigate()
+    // Current User Context
+    const {currentUser, setCurrentUser} = useContext(UserContext)
+
+
+    
+    
+    // fetching the registered users from db.json
+    useEffect(()=>{
+        axios.get("http://localhost:3000/data").then((res)=>{
+    
+          setRegisteredUsers(res.data)
+          //console.log(registeredusers)
+          
+    
+        }).catch((err) => {
+    
+          console.log('error')
+    
+        })
+    },[])
+
+    const handleSubmit = (e) => {
+        e.preventDefault()
+
+        const isRegistered = registeredusers.some(registeredusers => registeredusers.name === user.name && registeredusers.password === user.password)
+
+        const matchedUser = registeredusers.find(registeredusers => registeredusers.name === user.name && registeredusers.password === user.password)
+        
+            if(formaction === "login"){
+                    if(isRegistered)
+                    {   
+                        // If login process is successful update UserContext -> current_user and update isLogged which is passed by UserContext from parent component
+                        setCurrentUser({current_username: user.name, isLogged: true, current_id: matchedUser.id, current_password: matchedUser.password, current_favlist: matchedUser.fav})
+                        
+                        navigate("/popular")
+                        
+
+                    }
+                    else{
+
+                        
+                        alert("cannot log in...")
+                    }
+                }
+            else if(formaction ==="register"){
```

Comment:
My pickiness again 😆 
```suggestion
            else if(formAction ==="register"){
```